### PR TITLE
8314651: G1: Fix -Wconversion warnings in static fields of HeapRegion

### PR DESCRIPTION
**Pull request by albertnetymk**
Use unsigned type for heap-region-size related shifts in HeapRegion.
 
Test: tier1-3

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314651](https://bugs.openjdk.org/browse/JDK-8314651): G1: Fix -Wconversion warnings in static fields of HeapRegion (**Enhancement** - P4)

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15360/head:pull/15360` \
`$ git checkout pull/15360`

Update a local copy of the PR: \
`$ git checkout pull/15360` \
`$ git pull https://git.openjdk.org/jdk.git pull/15360/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15360`

View PR using the GUI difftool: \
`$ git pr show -t 15360`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15360.diff">https://git.openjdk.org/jdk/pull/15360.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15360#issuecomment-1686242519)

### CODE REVIEW

File: src/hotspot/share/gc/g1/c1/g1BarrierSetC1.cpp [hunk 1]
```diff
@@ -158,13 +158,13 @@ void G1BarrierSetC1::post_barrier(LIRAccess& access, LIR_Opr addr, LIR_Opr new_v
     __ logical_xor(xor_res, new_val, xor_res);
     __ move(xor_res, xor_shift_res);
     __ unsigned_shift_right(xor_shift_res,
-                            LIR_OprFact::intConst(HeapRegion::LogOfHRGrainBytes),
+                            LIR_OprFact::intConst(static_cast<jint>(HeapRegion::LogOfHRGrainBytes)),
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/c1/g1BarrierSetC1.cpp:161]: One could maybe argue that this should be a `checked_cast`, but given that the value bounds are known there is no point...

**Review by tschatzl — CHANGES_REQUESTED** [review]

File: src/hotspot/share/jvmci/vmStructs_jvmci.cpp [hunk 2]
```diff
@@ -817,7 +817,7 @@
 #if INCLUDE_G1GC
 
 #define VM_STRUCTS_JVMCI_G1GC(nonstatic_field, static_field) \
-  static_field(HeapRegion, LogOfHRGrainBytes, int)
+  static_field(HeapRegion, LogOfHRGrainBytes, uint32_t)
```
[the remark this answers is not in the document]
Comment (albertnetymk) [reply src/hotspot/share/jvmci/vmStructs_jvmci.cpp:820]: That was my original attempt, but JVMCI related tests failed with `uint`.

**Review by tschatzl — COMMENTED** [review]
Lgtm.

**Review by tschatzl — APPROVED** [review]

**Review by walulyai — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**